Author: Emilio Pozuelo Monfort <pochu@debian.org>

Even if logind is running (which is what haveSystemd() is checking), it may
fail to create a session, e.g. if the kernel doesn't have cgroups support.

If that's the case, XDG_SESSION_ID will not be set and we will fail to
start. Thus talk to logind if XDG_SESSION_ID is set, which is a sign that
logind actually created our session. Otherwise, fall back to ConsoleKit.

http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=729877

--- a/js/misc/loginManager.js
+++ b/js/misc/loginManager.js
@@ -117,7 +117,7 @@
  */
 function getLoginManager() {
     if (_loginManager == null) {
-        if (haveSystemd())
+        if (haveSystemd() && GLib.getenv('XDG_SESSION_ID'))
             _loginManager = new LoginManagerSystemd();
         else
             _loginManager = new LoginManagerConsoleKit();
